pyannote/audio/interactive/voice_activity_detection.py

```diff
@@ -0,0 +1,150 @@
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Dict, Iterable, Optional, Union
+
+import prodigy
+from prodigy.components.loaders import Audio as AudioLoader
+from utils import SAMPLE_RATE, chunks, normalize, to_audio_spans, to_base64
+
+from pyannote.audio.core.io import Audio
+from pyannote.audio.pipelines import VoiceActivityDetection
+from pyannote.core import Annotation, Segment
+
+
+def remove_base64(examples):
+    """Remove base64-encoded string if "path" is preserved in example."""
+    for eg in examples:
+        if "audio" in eg and eg["audio"].startswith("data:") and "path" in eg:
+            eg["audio"] = eg["path"]
+        if "video" in eg and eg["video"].startswith("data:") and "path" in eg:
+            eg["video"] = eg["path"]
+    return examples
+
+
+def voice_activity_detection_stream(
+    pipeline: VoiceActivityDetection, source: Path, chunk: float = 10.0
+) -> Iterable[Dict]:
+    """
+    Stream for pyannote.voice_activity_detection recipe
+    Applies (pretrained) speech activity detection and sends the results for
+    manual correction chunk by chunk.
+    Parameters
+    ----------
+    pipeline : VoiceActivityDetection
+        Pretrained speech activity detection pipeline.
+    source : Path
+        Directory containing audio files to process.
+    chunk : float, optional
+        Duration of chunks, in seconds. Defaults to 10s.
+    Yields
+    ------
+    task : dict
+        Prodigy task with the following keys:
+        "path" : path to audio file
+        "text" : name of audio file
+        "chunk" : chunk start and end times
+        "audio" : base64 encoding of audio chunk
+        "audio_spans" : speech spans detected by pretrained SAD model
+        "audio_spans_original" : copy of "audio_spans"
+        "meta" : additional meta-data displayed in Prodigy UI
+    """
+    raw_audio = Audio(sample_rate=SAMPLE_RATE, mono=True)
+
+    for audio_source in AudioLoader(source):
+
+        path = audio_source["path"]
+        text = audio_source["text"]
+        file = {"uri": text, "database": source, "audio": path}
+
+        duration = raw_audio.get_duration(file)
+        file["duration"] = duration
+
+        prodigy.log(f"RECIPE: detecting speech regions in '{path}'")
+
+        speech: Annotation = pipeline(file)
```

Comment:
Remember to only apply the pipeline on chunks to make things faster when running on CPU.